src/bacnet/basic/service/s_rp.c

```diff
@@ -132,9 +132,18 @@ uint8_t Send_Read_Property_Request(
     unsigned max_apdu = 0;
     uint8_t invoke_id = 0;
     bool status = false;
+#if BACNET_SEGMENTATION_ENABLED
+    uint8_t segmentation = 0;
+    uint16_t maxsegments = 0;
+#endif
 
     /* is the device bound? */
-    status = address_get_by_device(device_id, &max_apdu, &dest);
+    status = address_get_by_device(
```

Comment:
this change is not needed if `address_segment_get_by_device()` is added to address.c module and address_get_by_device() remains unchanged